include/openthread/platform/crypto.h

```diff
@@ -0,0 +1,496 @@
+    OT_CRYPTO_KEY_TYPE_HMAC, ///< Key Type: HMAC.
+} otCryptoKeyType;
+
+/**
+ * This enumeration defines the key algorithms.
+ *
+ */
+typedef enum
+{
+    OT_CRYPTO_KEY_ALG_VENDOR,       ///< Key Algorithm: Vendor Defined.
+    OT_CRYPTO_KEY_ALG_AES_ECB,      ///< Key Algorithm: AES ECB.
+    OT_CRYPTO_KEY_ALG_HMAC_SHA_256, ///< Key Algorithm: HMAC SHA-256.
+} otCryptoKeyAlgorithm;
+
+/**
+ * This enumeration defines the key usage types.
+ *
+ */
+typedef enum
+{
+    OT_CRYPTO_KEY_USG_EXPORT    = 1, ///< Key Usage: Key can be exported.
+    OT_CRYPTO_KEY_USG_ENCRYPT   = 2, ///< Key Usage: Vendor Defined.
+    OT_CRYPTO_KEY_USG_DECRYPT   = 4, ///< Key Usage: AES ECB.
+    OT_CRYPTO_KEY_USG_SIGN_HASH = 8, ///< Key Usage: HMAC SHA-256.
+} otCryptoKeyUsage;
+
+/**
+ * This enumeration defines the key storage types.
+ *
+ */
+typedef enum
+{
+    OT_CRYPTO_KEY_STORAGE_VOLATILE,   ///< Key Persistence: Key is volatile.
+    OT_CRYPTO_KEY_STORAGE_PERSISTENT, ///< Key Persistence: Key is persistent.
+} otCryptoKeyStorage;
+
+/**
+ * This datatype represents the key reference.
+ *
+ */
+typedef uint32_t otCryptoKeyRef;
+
+/**
+ * @struct otCryptoKey
+ *
+ * This structure represents the Key Material required for Crypto operations.
+ *
+ */
+
+OT_TOOL_PACKED_BEGIN
+struct otCryptoKey
+{
+    const uint8_t *mKey;       ///< Pointer to the buffer containing key. NULL indicates to use `mKeyRef`.
+    uint16_t       mKeyLength; ///< The key length in bytes (applicable when `mKey` is not NULL).
+    uint32_t       mKeyRef;    ///< The PSA key ref (requires `mKey` to be NULL).
+} OT_TOOL_PACKED_BEGIN;
+
+/**
+ *
+ * This structure represents the Key Material required for Crypto operations.
+ *
+ */
+typedef struct otCryptoKey otCryptoKey;
+
+/**
+ * Initialise the PSA module.
+ *
+ * @retval OT_ERROR_NONE          Successfully encrypted  @p aInput.
+ * @retval OT_ERROR_FAILED        Failed to encrypt @p aInput.
+ *
+ */
+otError otPlatCryptoInit(void);
+
+/**
+ * Import a key into PSA ITS.
+ *
+ * @param[out]  aKeyId            Reference to the key to be used for crypto operations.
+ * @param[in]   aKeyType          Key Type encoding for the key.
+ * @param[in]   aKeyAlgorithm     Key algorithm encoding for the key.
+ * @param[in]   aKeyUsage         Key Usage encoding for the key.
+ * @param[in]   aKeyPersistence   Key Persistence for this key
+ * @param[in]   aKey              Actual key to be imported.
+ * @param[in]   aKeyLen           Length of the key to be imported.
+ *
+ * @retval OT_ERROR_NONE          Successfully encrypted  @p aInput.
+ * @retval OT_ERROR_FAILED        Failed to encrypt @p aInput.
+ * @retval OT_ERROR_INVALID_ARGS  @p aInput or @p aOutput was set to NULL.
+ *
+ */
+otError otPlatCryptoImportKey(otCryptoKeyRef *     aKeyId,
+                              otCryptoKeyType      aKeyType,
+                              otCryptoKeyAlgorithm aKeyAlgorithm,
+                              int                  aKeyUsage,
```

Comment:
Well this can be any combination of the enumerations, like the key can be, exportable, used with encryption, decryption and so on. I think the better approach here is to typedef `otCryptoKeyUsage` to int and creating a unnamed enum for the values.